### PR TITLE
Addtodo

### PR DESCRIPTION
**Pull request by nitdevv**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+    EditComponent
   ],
   imports: [
-    BrowserModule
+    BrowserModule,
+    RouterModule.forRoot([
+      {path: '', pathMatch: 'full', redirectTo: 'dashboard'},
+      {path:'',component:DashboardComponent},
+      {path:'edit',component:EditComponent},
+      {path:'Edit/:id',component:EditComponent}
```
Comment (Rahul-Khera-Codes) [inline src/app/app.module.ts]: why 2 route for single component.

File: src/app/dashboard.component/dashboard.component.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<h3>TODOS</h3>
+<ul class="heroes">
+    <li *ngFor="let todo of todos"
+     [class.hidden]="showT(todo)">
```
Comment (upexcel) [inline src/app/dashboard.component/dashboard.component.html]: is there any use of this [class.hidden]="showT(todo)
if not remove this.

File: src/app/dashboard.component/todo.dashboard.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { HttpModule }    from '@angular/http';
+@Component({
+    selector: 'my-dashboard',
+    templateUrl: './dashboard.component.html',
+    styleUrls: ['../dashboard.component.css'],
+})
+export class DashboardComponent implements OnInit, OnChanges {
+    todos: Todo[] = [];
+    nametype: string;
+    show_todo: boolean;
```
Comment (Rahul-Khera-Codes) [inline src/app/dashboard.component/todo.dashboard.ts]: use camel case variable name

File: src/app/dashboard.component/todo.dashboard.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    }
+    ngOnInit() {
+        this.HeroService.getHeroes()
+            .then(todos => {
+                this.todos = todos
+            });
+    }
+
+    gotoDetail(todo: Todo) {
+     let link = ['Edit', { id: todo.id }];
```
Comment (Rahul-Khera-Codes) [inline src/app/dashboard.component/todo.dashboard.ts:31]: what is the use of this link ?

File: src/app/dashboard.component/todo.dashboard.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+     this.router.navigate(['/edit',todo.id]);
+ }
+    addTodo() {
+        let link = ['Edit'];
+        this.router.navigate(['/edit']);
```
Comment (upexcel) [inline src/app/dashboard.component/todo.dashboard.ts]: in your routing edit route need a id.  i didn't see any id in this this.router.navigate(['/edit']);

File: src/app/app.module.ts (hunk 1 above)
Comment (nitdevv) [reply src/app/app.module.ts]: sir when on add button i navigate it give error and i used to single route it don't give error thats why i used it

File: src/app/dashboard.component/dashboard.component.html (hunk 2 above)
Comment (nitdevv) [inline src/app/dashboard.component/dashboard.component.html]: sir for now it is not used but on later it is used. for now i remove it

File: src/app/dashboard.component/todo.dashboard.ts (hunk 3 above)
Comment (nitdevv) [reply src/app/dashboard.component/todo.dashboard.ts]: ok sir

File: src/app/dashboard.component/todo.dashboard.ts (hunk 4 above)
Comment (nitdevv) [reply src/app/dashboard.component/todo.dashboard.ts:31]: i simple create a variable

**Pull request merged**